workspaces/ui-v2/src/analytics/AnalyticsStore.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { FC, useContext, useEffect, useRef, useState } from 'react';
+import { OpticUIEvents } from '@useoptic/analytics/lib/optic-ui/OpticUIEvents';
+import { IOpticAnalyticsEvent } from '@useoptic/analytics/lib/interfaces';
+import invariant from 'invariant';
+import { useAppConfig } from '../optic-components/hooks/config/AppConfiguration';
+
+const packageJson = require('../../package.json');
+const clientId = `local_cli_${packageJson.version}`;
+
+const AnalyticsContext = React.createContext<
+  | {
+      trackEvent: OpticUIEvents;
+    }
+  | undefined
+>(undefined);
+
+type AnalyticsMetadata = {
+  clientId: string;
+  clientAgent: string;
+  apiName: string;
+};
+const defaultMetadata = {
+  clientId,
+  clientAgent: 'anon_id',
+  apiName: '',
+};
+
+export type AnalyticsStoreProps = {
```

Comment:
Exposed these fns which act as implementations. I think this would fit well as redux middleware.

Currently we need to access shared state through hooks, which isn't ideal since some hooks _may_ only be usable when the cli-server is running (meaning we either have to build the hooks to run in all contexts) or just access the data programatically here

I also considered building three implementations of AnalyticsStore, but we want to use the same `useAnalytics` hook - and to do that with rendered components is a little challenging / bad for render optimization

Reply:
I thought as long as the context provider was the same, the useAnalytics hook could be ignorant of the value provided by the different store impls?

Reply:
Correct - the key part is the context provider needs to be the same. If we need to call different hooks conditionally in different contexts, we'd probably need to mount different components in the `AnalyticsStore` which can get messy

In this, I'm just pulling out any hook related content and changing them work programatically